daemon/cmd/kube_proxy_replacement.go

```diff
@@ -307,6 +307,26 @@ func detectNativeDevices(strict bool) {
 // replacement after all devices are known.
 func finishKubeProxyReplacementInit(isKubeProxyReplacementStrict bool) {
 	if option.Config.EnableNodePort {
+		// Check whether the DirectRoutingDevice (if specified) is defined within devices
+		if option.Config.DirectRoutingDevice != "" {
+			directDev := option.Config.DirectRoutingDevice
+			directDevFound := false
+			for _, iface := range option.Config.Devices {
+				if iface == directDev {
+					directDevFound = true
+					break
+				}
+			}
+
+			if !directDevFound {
+				// docs state that:
```

Comment:
Nit: Not sure I get why we need to specifically call out that the docs state this?

Reply:
That's a good point, but I wanted to document the reasoning for choosing this behavior but it does not make sense.
I'll remove it.